### PR TITLE
Add DevTools memory test

### LINKED ISSUE #44013: Persist the memory profile timeline
As discussed with @terrylucas and @cobblest earlier, we'd like to persist the memory profile timeline (total usage without components breakdown is sufficient) for automated performance benchmarks (e.g., complex_layout_scroll_perf__memory in https://flutter-dashboard.appspot.com/benchmarks.html). It seems that DevTools can provide such a tool without too much work.

Specifically, it would be nice to have a command line tool so we can:
1. Start that tool in command line
2. Start a Flutter driver test (e.g., `flutter drive test_driver/scroll_perf.dar` inside `flutter/dev/benchmarks/complex_layout`)
3. Stop the tool when the driver test finishes
4. The tool exports a json file that represents the memory timeline between step 1 and 3.

Related issue: https://github.com/flutter/flutter/issues/40406

### PR DESCRIPTION
**Pull request by liyuqian**
This allows us to calculate the max memory usage during a driver test, and closes https://github.com/flutter/flutter/issues/44013.

Although DevTools only provides memory samples with 1 second granularity, the max memory metrics we get from them seems to be quite stable. The following is the metrics distribution for 10 local runs. The `maxAdbTotal` has a standard deviation of 3.76% and the `maxRss` has a standard deviation of 2.2%.

|  maxAdbTotal | maxRss (KB) |
| --- | --- |
|  65566 | 99635.2 |
|  64406 | 98213.888 |
|  63470 | 97325.056 |
|  60051 | 97783.808 |
|  60219 | 98017.28 |
|  59434 | 93212.672 |
|  59712 | 93593.6 |
|  62793 | 97071.104 |
|  64230 | 98119.68 |
|  64542 | 98631.68 |

Screen Shot 2020-04-23 at 12 06 16 PM

### CODE REVIEW

File: dev/devicelab/manifest.yaml [hunk 1]
```diff
@@ -279,6 +279,12 @@ tasks:
     stage: devicelab
     required_agent_capabilities: ["mac/android"]
 
+  complex_layout_scroll_perf__devtools_memory:
+    description: >
+      Measures memory usage of the scroll performance test using DevTools.
+    stage: devicelab
+    required_agent_capabilities: ["mac/android"]
```
Comment (dnfield) [inline dev/devicelab/manifest.yaml:286]: Can we run this on Linux instead of mac? We have more Linux machines AFAIK.  @digiter in case I'm wrong.

Comment (digiter) [reply dev/devicelab/manifest.yaml:286]: They are roughly the same, but I prefer Linux ones as they are more stable.

Comment (liyuqian) [reply dev/devicelab/manifest.yaml]: Done.

**Review by dnfield — APPROVED** [review]
LGTM

File: dev/devicelab/manifest.yaml [hunk 2]
```diff
@@ -279,6 +279,12 @@ tasks:
     stage: devicelab
     required_agent_capabilities: ["mac/android"]
 
+  complex_layout_scroll_perf__devtools_memory:
+    description: >
+      Measures memory usage of the scroll performance test using DevTools.
+    stage: devicelab
+    required_agent_capabilities: ["linux/android"]
```
Comment (dnfield) [inline dev/devicelab/manifest.yaml:286]: oh, one other nit: may be worth starting this out as `flaky` so if it fails initially you can fix it without needing to revert.

Comment (dnfield) [reply dev/devicelab/manifest.yaml:286]: (or otherwise block the tree)

**Pull request merged**